controllers/glance_controller.go

```diff
@@ -766,3 +781,48 @@ func (r *GlanceReconciler) generateServiceConfigMaps(
 
 	return nil
 }
+
+// ensureRegisteredLimits - create registered limits in keystone that will be
+// used by glance to enforce per-tenant quotas
+func (r *GlanceReconciler) ensureRegisteredLimits(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *glancev1.Glance,
+) (string, error) {
+
+	quota := map[string]int{
+		"image_count_uploading": instance.Spec.Quotas.ImageCountUpload,
+		"image_count_total":     instance.Spec.Quotas.ImageCountTotal,
+		"image_stage_total":     instance.Spec.Quotas.ImageStageTotal,
+		"image_size_total":      instance.Spec.Quotas.ImageSizeTotal,
+	}
+
+	// get admin
+	var err error
+	var limit string = ""
+	keystoneAPI, err := keystonev1.GetKeystoneAPI(ctx, h, instance.Namespace, map[string]string{})
+
+	if err != nil {
+		return limit, err
+	}
+	o, _, err := keystonev1.GetAdminServiceClient(ctx, h, keystoneAPI)
+	if err != nil {
+		return limit, err
+	}
+	for lName, lValue := range quota {
+		r.Log.Info(fmt.Sprintf("Registering limit %s", lName))
+		defaultRegion := o.GetRegion()
+		m := openstack.RegisteredLimit{
+			RegionID:     defaultRegion,
+			ServiceID:    instance.Status.ServiceID,
+			Description:  "default",
```

Comment:
I think we can skip this or generate value like, "default limit for " + lNmae

Reply:
+1 on a description like the one you mentioned! I'll do that in the next PS

Reply:
Done